endpoint.go

```diff
@@ -890,6 +890,14 @@ func CreateOptionPortMapping(portBindings []types.PortBinding) EndpointOption {
 	}
 }
 
+// CreateOptionDNS function returns an option setter for dns entry option to
+// be passed to container Create method.
+func CreateOptionDNS(dns []string) EndpointOption {
+	return func(ep *endpoint) {
+		ep.generic[netlabel.DNSServers] = dns
+	}
+}
+
```

Comment:
@mavenugo With the current libnetwork model you can have an endpoint created before the sandbox is created.  We don't allow updating endpoints after they have been created.  We need the DNS information before endpoint creation and the current model doesn't guarantee it.